src/main/kotlin/se/uulm/snowballr/backend/repository/ProjectTableRepo.kt

```diff
@@ -62,6 +74,21 @@ interface IProjectTableRepo {
             ProjectStatus.PROJECT_STATUS_ACTIVE_LOCKED,
         ),
     ): List<Project>
+
+    /**
+     * Updates an existing project in the database with the provided new information.
+     * The following fields can be updated:
+     * - name
+     * - status
+     * - similarity_threshold
+     * - snowballing_type
+     * - review_maybe_allowed
+     *
+     * @param request The update request containing the new project details, such as the new name.
+     * @param activeLocked True, if the project is of status ACTIVE_LOCKED (only the name and status can be updated then).
+     * @return The updated [Project] object reflecting the changes from the [request].
+     */
+    suspend fun updateProject(request: ProjectOuterClass.Project.Update, activeLocked: Boolean): Project
```

Comment:
Use terms with `is`, `was`, `does` to name boolean values.
```suggestion
    suspend fun updateProject(request: ProjectOuterClass.Project.Update, isActiveLocked: Boolean): Project
```